talisker/testing.py

```diff
@@ -140,12 +151,58 @@ class ServerProcessError(Exception):
 
 class ServerProcess(object):
     """Context mananger to run a server subprocess """
-    def __init__(self, cmd):
+    def __init__(self, cmd, env=None):
         self.cmd = cmd
+        self.env = env
         self.output = []
         self.ps = None
         self._log = None
 
+    def start(self):
+        self.output_file = temp_file()
+        self.reader = open(self.output_file.name, 'r')
+        self.ps = subprocess.Popen(
+            self.cmd,
+            bufsize=0,
+            stdout=self.output_file,
+            stderr=subprocess.STDOUT,
+            stdin=subprocess.PIPE,
+            universal_newlines=True,
+            env=self.env
+        )
+        try:
+            self.check()
+        except Exception:
+            self.close(error=True)
+            raise
+
+    def close(self, error=False):
+        """Clean up process."""
+        if not self.finished:
+            self.ps.terminate()
+            self.ps.wait()
+
+        if self.output_file and not self.output_file.closed:
+            self.output_file.flush()
```

Comment:
Does either this flush or the one in `readline` actually do anything?  This flushes the stream's write buffers, but it can't flush write buffers from some other process, and this file object is only ever written to by other processes.

Reply:
Right - the process is definitely dead by now, and thus any output flushed anyway. Removed.